quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoAssetsProcessor.java

```diff
@@ -385,4 +391,14 @@ private KogitoGeneratedClassesBuildItem indexBuildItems(KogitoBuildContext conte
 
         return new KogitoGeneratedClassesBuildItem(kogitoIndexer.complete(), generatedClasses);
     }
+
+    static class IsDumpingFiles implements BooleanSupplier {
+        LaunchMode launchMode;
+
+        @Override
+        public boolean getAsBoolean() {
+            return launchMode == LaunchMode.DEVELOPMENT && ConfigProvider.getConfig().getOptionalValue("kogito.quarkus.codegen.dumpFiles", Boolean.class).orElse(true);
```

Comment:
I'm not a Quarkus dev mode expert but how `launchMode` value is injected? "Automatically" by Quarkus?
As far as I understand [from here](https://quarkus.io/guides/lifecycle#launch-modes) the code should use `current()` to fetch the value
```suggestion
        @Override
        public boolean getAsBoolean() {
            return LaunchMode.current() == LaunchMode.DEVELOPMENT && ConfigProvider.getConfig().getOptionalValue("kogito.quarkus.codegen.dumpFiles", Boolean.class).orElse(true);
```

Reply:
I changed it while you were commenting